File: unit2/p2.js
// Unit2.5 find the second largest
// answer
let arr = [5, 8, 6]
let max = -Infinity
let max2 = -Infinity
for (let i=0; i<arr.length; i++) {
  if (arr[i] > max) {
    max2 = max
    max = arr[i]
  } else if (arr[i] > max2) {
    max2 = arr[i]
  }
}
console.log(max, max2)

// Unit2.6 Upper the string
let str = "You better run now! az"
// solution 1
console.log(str.toUpperCase())
// solution 2
let ans1 = ""
for (let i=0; i<str.length; i++) {
  let code = str.charCodeAt(i)
  // console.log(code)
  if (code >= 97 && code <= 122) {
    ans1 += String.fromCharCode(code - 32)
  } else {
    ans1 += str[i]
  }
}
console.log(ans1)
// solution 3
let str3 = "Hi az"
let ans3 = ''
for(let i=0; i<str3.length; i++) {
  if (str3[i] >= 'a' && str3[i] <= 'z') {
    ans3 += String.fromCharCode(
      str3.charCodeAt(i) - 32
    )
  } else {
    ans3 += str3[i]
  }
}
console.log(ans3)

// Unit2.7 Delete specifit letter in a string
let origin = "Jumpy"
let letter = "y"
let ans = ""
for (let i=0; i<origin.length; i++) {
  if (origin[i] != letter)  {
    ans = ans + origin[i]
  }
}
console.log(ans)

// Unit2.8 Project 2
// 1. Find the second smallest
// 2. lower case to upper case and vise versa
// 3. Print the factor numbers

File: unit2/p2.py
### Unit2.5 find the second largest

arr = [2, 7, 5, 8, 4]
max = arr[0]
max2 = 0

for i in arr:
    # print(i)
    if i > max:
        max2 = max
        # print('max2=', max2)
        max = i
        # print('max=', max)
    elif i > max2:
        max2 = i
        # print('max2=', max2)
        # print('max=', max)

print("final max2 =", max2)

####

first = -2147483648
second = -2147483648
for i in range(0, len(arr)):
    if (arr[i] > first):
        second = first
        first = arr[i]
    elif (arr[i] > second and arr[i] != first):
        second = arr[i]

print("The second largest element is", second)

### Unit2.6 Capitalize all the letters in a string
# ASCII https://zh-yue.wikipedia.org/wiki/ASCII
# A = 65, Z = 90, a = 97, z = 122
# ASCII to int: ord('a') = 97
# int to string str(chr(97)) = 'a'
# Input: Wolverine
# OUTPUT: WOLVERINE

## Psuedo code
# tell if it is lower case
# if not - omit
# if yes - transfer

origin = "Wolverine"
ans = ""
for l in origin:
    # if it is lower case
    if ord(l) <= 122 and ord(l) > 97:
        # if yes: minus 32 to upper case
        ans = ans + str(chr(ord(l) - 32))
    # if not: as original
    else:
        ans = ans + l
print(ans)

# OR using .upper() function
print(origin.upper())

### Unit2.7 Delete specifit letter in a string

str1 = "apple"
letter1 = "l"
str2 = ""
for l in str1:
    if l != letter1:
        # if yes: add it, if not, omit it
        str2 = str2 + l
print(str2)


### Unit2.8 Project 2
# 1. Find the second smallest
# 2. lower case to upper case and vise versa
# 3. Print the factor numbers
